src/sota_tools/deploy.cc

```diff
@@ -57,12 +57,18 @@ bool UploadToTreehub(const OSTreeRepo::ptr &src_repo, const ServerCredentials &p
 
 bool OfflineSignRepo(const ServerCredentials &push_credentials, const std::string &name, const OSTreeHash &hash,
                      const std::string &hardwareids) {
-  if (!boost::filesystem::is_directory(boost::filesystem::path("./tuf/aktualizr"))) {
-    std::string init_cmd("garage-sign init --repo aktualizr --credentials ");
-    if (system((init_cmd + push_credentials.GetPathOnDisk().string()).c_str()) != 0) {
-      LOG_ERROR << "Could not initilaize tuf repo for sign";
-      return false;
-    }
+  const boost::filesystem::path local_repo{"./tuf/aktualizr"};
+
+  // OTA-682: Do NOT keep the local tuf directory around in case the user tries
+  // a different set of push credentials.
+  if (boost::filesystem::is_directory(local_repo)) {
```

Comment:
Should it just be a temporary directory, since it's pretty ephemeral?

Reply:
That would be great, but that's up to garage-sign. As far as I can tell, `garage-sign init` only gives you the `--repo` option, but whatever you provide is actually used as a subdirectory of `$PWD/tuf/`. I guess we could change the working directory before running these commands. Is that worth it?

Reply:
Ok, makes sense.